### PR TITLE
Max

### LINKED ISSUE #18: buy crowncard max
make it so `buy crowncard max` buys as many crowncards as the user can afford

### PR DESCRIPTION
**Pull request by amlubinger**
Closes #18 

`t?buy crowncards max` will purchase the maximum amount of crown gift cards based on the user's coin balance.

Tested with:
- lowercase max
- capital MAX
- balance < 110
- balance = 110*n
- balance > 110*n
- specifying numerical quantity
- not specifying quantity

### CODE REVIEW

File: util/shop/shop.js [hunk 1]
```diff
@@ -125,11 +125,18 @@ const royalty = (userId, callback) => {
...
-      price *= Number.parseInt(quantity, 10);
+    let quantity = 1;
+    if (quant && quant.toLowerCase() === 'max') {
+      quantity = Math.floor(data.getCoins(userId) / price);
+      if (quantity === 0) quantity = 1;
+    } else if (!Number.isNaN(Number.parseInt(quant, 10))) {
+      quantity = Number.parseInt(quant, 10);
+    }
+    if (enabledPowerups[item].quantified) {
+      price *= quantity;
```
Comment (ajivoin) [inline util/shop/shop.js:139]: can you move everything inside of `if (enabledPowerups[item].quantified)` so this code only runs when needed?

**Review by ajivoin — CHANGES_REQUESTED** [review]

File: util/shop/shop.js (hunk 1 above)
Comment (amlubinger) [reply util/shop/shop.js:139]: sure thing jelly bean

Comment (ajivoin) [reply util/shop/shop.js:139]: did you do the same testing after you made this change too? I know it shouldn't mess anything up but stranger things have happened

Comment (amlubinger) [reply util/shop/shop.js:139]: yes

**Review by ajivoin — APPROVED** [review]

**Pull request merged**